src/unistd.rs

```diff
@@ -356,6 +375,13 @@ pub fn getcwd() -> Result<PathBuf> {
     }
 }
 
+// According to the POSIX specification, -1 is used to indicate that
```

Comment:
Because you're referring to `uid_t` and `gid_t` which aren't used in this function. The description to this should be more broad (not referring to those specific use cases) and then you can put comments stating why you need to use this function at the call site that refers to specific variables.